File: src/main/java/com/nao4j/subtrol/service/slice/SubscriptionSlicer.java
package com.nao4j.subtrol.service.slice;

import com.nao4j.subtrol.document.internal.ExactPeriod;
import com.nao4j.subtrol.document.internal.RightOpenPeriod;

import java.util.Collection;

public interface SubscriptionSlicer {

    Collection<ExactPeriod> slice(RightOpenPeriod subscriptionPeriod, ExactPeriod calculationPeriod, long stepSize);

}

File: src/main/kotlin/com/nao4j/subtrol/service/slice/SubscriptionSlicer.kt
package com.nao4j.subtrol.service.slice

import com.nao4j.subtrol.model.ExactPeriod
import com.nao4j.subtrol.model.RightOpenPeriod

interface SubscriptionSlicer {

    fun slice(
        subscriptionPeriod: RightOpenPeriod,
        calculationPeriod: ExactPeriod,
        stepSize: Long
    ): Collection<ExactPeriod>

}
